fabric-content-registries-v1/src/main/java/net/fabricmc/fabric/api/content/registry/v1/item/UseTickable.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.content.registry.v1.item;
+
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+
+public interface UseTickable {
+	/**
+	 * Called each tick when the item is being used.
+	 * @param stack The {@link ItemStack} being used
+	 * @param player The {@link PlayerEntity} using the item
+	 * @param time The amount of time in ticks that the item is being used
+	 */
+	void useTick(ItemStack stack, PlayerEntity player, int time);
```

Comment:
is this needed?

Reply:
yes. my custom food spawn enderdragons when being eaten

Reply:
bit of a nieche usecase. find 2 more reasons to have it and ill resolve the conversation

Reply:
my food makes the player immune to attackers when being eaten

Reply:
my item spawns fireballs when being used

Reply:
my item sends packets to the server when being used

Reply:
this is good usecase

Reply:
making the player immune and sending packets are not a good usecase?
but spawning fireballs and enderdragons are? *fr* too?

Reply:
the opposite

Reply:
that's 2 and 4
not 1 and 3
1 and 3 are the same

Reply:
yes
always fix my confusion